### PR TITLE
build: add its own include path to each library

### PR DESCRIPTION
**Pull request by federico-sysdig**
**What type of PR is this?**
/kind cleanup
/kind feature

**Any specific area of the project related to this PR?**
/area build

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:
Libraries of `falcosecurity/libs` do not bring their own include path with each CMake target. This forces clients to set this include path on their own, possibly getting it wrong or failing to track it when it changes. This PR fills this gap, thus simplifying usage of the libraries.
Also, the list of libraries linked by many targets was missing the PRIVATE/PUBLIC property. This has been fixed by using PRIVATE.
 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Using libraries `sinsp` or `scap` from a client CMake project does not require to specify any specific include path. The two targets will bring their own include path once they are referenced in the `target_link_libraries` CMake statement.
```

### CODE REVIEW

File: userspace/libpman/CMakeLists.txt [hunk 1]
```diff
@@ -48,8 +48,6 @@ set(PMAN_LINK_LIBRARIES
     scap_platform
 )
 
-set(PMAN_DEPENDENCIES scap_platform)
-
 if(USE_BUNDLED_MODERN_BPF)
     list(APPEND
         PMAN_DEPENDENCIES
```
[the remark this answers is not in the document]
Comment (federico-sysdig) [reply userspace/libpman/CMakeLists.txt]: I-ve removed the dependencies already referenced in `target_link_libraries` as they are redundant. I can't really remove the `add_dependencies` statement altogether here as it references something different, but I can definitely remove the `PMAN_DEPENDENCIES` variable. I agree the script is a bit simpler without it.

**Review by Andreagit97 — DISMISSED** [review]
/approve

**Comment by Andreagit97** [comment]
This seems almost ready I would try this for 0.14.0

**Comment by Andreagit97** [comment]
/hold
I will rerun the CI just to be sure it's ok

**Comment by FedeDP** [comment]
Needs to be rebased on top of latest master to fix ci.

**Review by incertum — APPROVED** [review]
/approve

**Comment by incertum** [comment]
ohh we need another rebase, fun!

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**